src/api/v1/db/oracledb/doughs-dao.js

```diff
@@ -0,0 +1,149 @@
+import config from 'config';
+import _ from 'lodash';
+
+import { serializeDoughs } from 'api/v1/serializers/doughs-serializer';
+import { openapi } from 'utils/load-openapi';
+import { getConnection } from './connection';
+
+const { endpointUri } = config.get('server');
+const doughsGetParameters = openapi.paths['/doughs'].get.parameters;
+
+/**
+ * an object mapping DoughRecipe property names to DOUGH table column names
+ * @constant
+ * @type {object}
+ */
+const doughColumnNames = {
+  id: 'ID',
+  name: 'NAME',
+  gramsFlour: 'GRAMS_FLOUR',
+  gramsWater: 'GRAMS_WATER',
+  flourType: 'FLOUR_TYPE',
+  waterTemp: 'WATER_TEMP',
+  gramsYeast: 'GRAMS_YEAST',
+  gramsSalt: 'GRAMS_SALT',
+  gramsSugar: 'GRAMS_SUGAR',
+  gramsOliveOil: 'GRAMS_OLIVE_OIL',
+  bulkFermentTime: 'BULK_FERMENT_TIME',
+  proofTime: 'PROOF_TIME',
+  specialInstructions: 'SPECIAL_INSTRUCTIONS',
+};
+
+/**
+ * a list of SQL aliases mapping DOUGH table column names to DoughRecipe properties,
+ *  generated from the DoughColumnNames object
+ * @constant
+ * @type {string}
+ */
+const doughColumnAliases = _.toPairs(doughColumnNames)
+  .map(([propertyName, columnName]) => `${columnName} AS "${propertyName}"`)
+  .join(', ');
+
+/**
+ * Turns a string containing conditional statements like `NAME = :name AND WATER_TEMP = :waterTemp`
+ *  into a fully fledged sql statement
+ * @param {string} filters
+ * @returns {string} The assembled query
+ */
+const doughsGetQuery = (filters) => `SELECT ${doughColumnAliases} FROM DOUGHS ${filters ? `WHERE ${filters}` : ''}`;
+
+
+/**
+ * Removes the "filter[]" wrapper from a filter parameter
+ * @param {string} filterName
+ * @returns {string} the filter name with 'filter' and brackets removed
+ */
+const normalizeFilter = (filterName) => {
+  const filterTextRegex = /filter\[(.*)\]/g;
```

Comment:
I'm wondering if there's a way to define filterTextRegex as a module-level constant so it can be used elsewhere, but when I was testing it seemed like it had some kind of internal state that was preventing it from being reused between function calls.

Reply:
I don't use it anywhere else right now. My main concern is that this is parsing the regex every time the function is called, and that if we *did* want to use it somewhere else there's currently no way to do that.

Reply:
Sounds fair. There are a few ways we can set global variables (e.g. https://stackoverflow.com/questions/9765215/global-variable-in-app-js-accessible-in-routes), but we should be very careful to set global variables in case adding unnecessary complexity to the project.